CountingSort/app_resources/prefix_sum_shader.comp.hlsl

```diff
@@ -1,8 +1,39 @@
 #include "nbl/builtin/hlsl/bda/__ptr.hlsl"
 #include "nbl/builtin/hlsl/sort/counting.hlsl"
+#include "nbl/builtin/hlsl/sort/common.hlsl"
 
 [[vk::push_constant]] nbl::hlsl::sort::CountingPushData pushData;
 
+struct PtrAccessor
+{
+    static PtrAccessor createAccessor(uint64_t addr)
+    {
+        PtrAccessor ptr;
+        ptr.addr = addr;
+        return ptr;
+    }
+
+    uint32_t get(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().load();
+    }
+
+    void set(uint64_t index, uint32_t value)
+    {
+        bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().store(value);
+    }
+
+    uint32_t atomicAdd(uint64_t index, uint32_t value)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().atomicAdd(value);
+    }
+
+    uint64_t addr;
```

Comment:
you want a separate source and dest address, so you can load and store in parallel when scattering